### PR TITLE
WIP : TW-247 : Improve members autocompletion input row

### PR DESCRIPTION
**Pull request by imGok**
#247 

https://github.com/linagora/twake-on-matrix/assets/48354990/457bd574-6257-4c7a-a146-1ff382d9324c

### CODE REVIEW

File: lib/pages/chat/input_bar.dart [hunk 1]
```diff
@@ -152,7 +152,7 @@ class InputBar extends StatelessWidget {
         }
       }
     }
-    final userMatch = RegExp(r'(?:\s|^)@([-\w]+)$').firstMatch(searchText);
+    final userMatch = RegExp(r'(?:\s|^)@([-\w]*)$').firstMatch(searchText);
```
Comment (hoangdat) [inline lib/pages/chat/input_bar.dart:155]: can you create the const name for regex, easier to understand what use for?

File: lib/pages/chat/input_bar.dart [hunk 2]
```diff
@@ -265,7 +265,7 @@ class InputBar extends StatelessWidget {
     if (suggestion['type'] == 'user') {
       insertText = '${suggestion['mention']!} ';
       startText = replaceText.replaceAllMapped(
-        RegExp(r'(\s|^)(@[-\w]+)$'),
+        RegExp(r'(\s|^)(@[-\w]*)$'),
```
Comment (hoangdat) [inline lib/pages/chat/input_bar.dart:268]: idem

**Review by drminh2807 — APPROVED** [review]

**Review by nqhhdev — APPROVED** [review]

**Review by sherlockvn — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Comment by hoangdat** [comment]
the demo not show anything related to your change. Can you upload another demo related to your logic?

**Comment by imGok** [comment]
> the demo not show anything related to your change. Can you upload another demo related to your logic?

we can now see all users when typing only "@"

**Comment by hoangdat** [comment]
What happen if this group chat have hundred of people? @imgok

**Pull request merged**